timelock-agent/src/main/java/com/palantir/timelock/paxos/PersistedSchemaVersion.java

```diff
@@ -30,6 +30,9 @@
     private static final String ONLY_ROW = "r";
     private final Jdbi jdbi;
 
+    @SuppressWarnings("OptionalUsedAsFieldOrParameterType") // better than setting to null...
+    private Optional<Long> persistedVersion = Optional.empty();
```

Comment:
AFAICT this is used:
1. on startup
2. for health checks

Obviously, this will still be checked on startup (it needs to load the first time, of course), but it makes the health check fairly redundant. I think instead we should probably just cache this for a certain amount of time. The health checks will check it every 30 seconds (I _think_), but maybe we can cache this for say 5 minutes? That, or we just retry here. We could just use a memoized supplier if we want to cache for a few minutes at a time (probably easier than hand-rolling something), although maybe I'm missing some context around why we want to cache this in the first place.

Reply:
Or, if we fail to read it, we report an empty value, and internally use that to WARN instead of ERROR, or something like that.

Reply:
We'd want to cache it in order to avoid health check flakes when we fail to communicate with the DB.

The purpose of the health check is not to test that we can talk to the DB where our schema version is stored, but that the schema version is up to date; therefore, I don't think it's worth periodically refetching the value, because it will never change except through this class.

Using a memoized supplier would work reasonably well, although we'd have to update it when we bump the schema version.

Reply:
Discussed offline, but the schema versions are stored locally on each node (duh) and thus this isn't really a concern - we don't expect this to change under us